storage-proofs-update/src/circuit.rs

```diff
@@ -97,69 +96,48 @@ impl PublicInputs {
     }
 }
 
-pub struct ChallengeProof<TreeR>
+#[derive(Clone)]
+pub struct ChallengeProof<U, V, W>
```

Comment:
Note: make ESU structures generic over TreeR artities `U, V, W` rather than `TreeR: MerkleTreeTrait` because:
- Makes clarifies that TreeR always uses an `LCStrore` store
- Being generic over `TreeR` makes it difficult to limit `TreeR::Hasher` to `PoseidonHasher<F>`; would need a `trait TreeRHasher` to be implemented for all `PoseidonHasher<F>` (which will require messier trait bounds)